pkgs/native_toolchain_c/lib/src/cbuilder/cbuilder.dart

```diff
@@ -188,7 +189,7 @@ class CBuilder extends CTool implements Builder {
           name: assetName!,
           file: libUri,
           linkMode: linkMode,
-          os: config.targetOS,
+          os: config.codeConfig.targetOS,
```

Comment:
Here we pass `os: ` and `architecture: ` separately. It may? be more idiomatic to pass an `Abi` object here instead.

Reply:
If we go that direction, we should also have `config.codeConfig.targetAbi` instead of OS and architecture. But that might make `config.codeConfig.android.ndkVersion` less obvious again. Anyway, let's discuss that in a separate issue.